File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int nLen = Integer.bitCount(n);
        for(int i = n+1 ; i <= 1000000; i++) {
            int bigLen = Integer.bitCount(i);
            if(nLen == bigLen) {
                answer = i;
                break;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.js
function solution(n) {
    var answer = 0;
   
    
    const count1 = n.toString(2).split('').filter((v)=> v==='1').length;
    
    while (true) {
        n++;
        let count2 = n.toString(2).split('').filter((v)=> v==='1').length;
        if (count1 === count2) break;
        
    }
    return n;
}